packages/material-ui/src/styles/createTheme.js

```diff
@@ -1,43 +1,21 @@
 import { deepmerge } from '@material-ui/utils';
 import { generateUtilityClass } from '@material-ui/unstyled';
-import createBreakpoints from './createBreakpoints';
-import createMixins from './createMixins';
+import { createTheme as systemCreateTheme } from '@material-ui/system';
 import createPalette from './createPalette';
 import createTypography from './createTypography';
-import shadows from './shadows';
-import shape from './shape';
-import createSpacing from './createSpacing';
-import createTransitions from './createTransitions';
 import zIndex from './zIndex';
 
 function createTheme(options = {}, ...args) {
-  const {
-    breakpoints: breakpointsInput = {},
-    mixins: mixinsInput = {},
-    palette: paletteInput = {},
-    spacing: spacingInput,
-    transitions: transitionsInput = {},
-    typography: typographyInput = {},
-    ...other
-  } = options;
+  const { palette: paletteInput = {}, typography: typographyInput = {}, ...other } = options;
 
   const palette = createPalette(paletteInput);
-  const breakpoints = createBreakpoints(breakpointsInput);
-  const spacing = createSpacing(spacingInput);
+  const systemTheme = systemCreateTheme(options);
 
   let muiTheme = deepmerge(
+    systemTheme,
     {
-      breakpoints,
-      direction: 'ltr',
-      mixins: createMixins(breakpoints, spacing, mixinsInput),
-      components: {}, // Inject component definitions
       palette,
-      // Don't use [...shadows] until you've verified its transpiled code is not invoking the iterator protocol.
-      shadows: shadows.slice(),
       typography: createTypography(palette, typographyInput),
-      spacing,
-      shape: { ...shape },
-      transitions: createTransitions(transitionsInput),
```

Comment:
Not sure about the default value. Aren't the transition values are specific to Material Design? Same for the shadows, border radius, mixins, spacing, breakpoints.

Reply:
I'm not sure that any of them should be in the system. But it's not all black and white; my attempt at ranking by order of what MD is the most unique about and the system should likely no include:

1. mixins very specific to the AppBar
2. shadows very specific to MD
3. border-radius, specific to MD
4. transition, specific to MD
5. breakpoints, somewhat generic 
6. spacing, somewhat generic (TW uses 4px, we use 8px, BS uses 8px)

Reply:
> @material-ui/system: parsed: +7.94% , gzip: +7.85%

To be careful because the larger the system is, the less likely it will be used by third parties. What we currently have: https://bundlephobia.com/result?p=@material-ui/system@5.0.0-alpha.35.

Reply:
> To be careful because the larger the system is, the less likely it will be used by third parties.

It is still not viable to cater to these people. Tree-shaking is a thing. If you want to live in the past with old technology then stay there. We're moving forwards where overall package size is mostly irrelevant.

Reply:
My opinion on the matter:

> ~mixins very specific to the AppBar~ - agree, will remove it
~shadows very specific to MD~ - agree, will remove it
border-radius, specific to MD - I will think it's good default, would rather leave it, other can override
~transition, specific to MD~ - agree, will remove it
breakpoints, somewhat generic - agree would leave it
spacing, somewhat generic (TW uses 0.25rem, we use 8px, BS uses 0.5rem) - agree, would leave it

Regarding the typings, I would leave all values in the `ThemeOptions` and as optional in the `Theme`, and in the core we can make the typings stricter, basically as they are in the moment. What do you think?

Regarding the bundle size, we are extending this package to be the styling package for building a custom design system, so I don't think we can compare with the previous implementation, but agree that we should be careful around not adding too many things which are not generic enough.

Reply:
@mnajdova Sounds fair.

---

> Tree-shaking

@eps1lon In the current approach, there is no tree shaking possible when importing `Box` or `styled`, we force the default theme. AFAIK, these two are meant to be the most popular modules used from *@material-ui/system*. 

Do you mean that you would make the default version of the helper with an empty theme and allow developers to cherry-pick which baseline theme they want? It could be a thing

> If you want to live in the past with old technology then stay there.

I assume "you" is targeted at the audience that is excessively worried about bundle size. The number of developers that were complaining about bundle size was relatively low in our last survey, maybe this audience is marginal. Assuming it's not a filtering bias, it supports the idea that we can take some liberties with bundle size and not care too much. For instance, chakra having all the styles of the components inside a non-tree-shakeable theme.